### PR TITLE
Return response on message channels

### PR DESCRIPTION
**Pull request by okjodom**
Instead of panicking on channel failures in `GatewayRpcSender`, we return error responses, thus adding resiliency to our messaging pipeline between CLN node and the gateway.

### CODE REVIEW

File: gateway/ln-gateway/src/rpc/mod.rs [hunk 1]
```diff
@@ -34,11 +35,20 @@ impl GatewayRpcSender {
...
+        receiver
             .await
-            .expect("failed to send over channel");
-        Ok(receiver.await.expect("Failed to send over channel")?)
+            .unwrap_or_else(|_| {
+                Err(LnGatewayError::Other(anyhow!(
+                    "Failed to receive response over channel"
+                )))
+            })
+            .map_err(|e| e.into())
```
Comment (elsirion) [inline gateway/ln-gateway/src/rpc/mod.rs:51]: This code doesn't feel super idiomatic but I can't find any better way of expressing what it does. Maybe @dpc as an idea? Not a blocker, would just be neat to have in my mental toolbox.

**Review by elsirion — APPROVED** [review]

**Comment by elsirion** [comment]
Related to https://github.com/fedimint/fedimint/pull/1151#pullrequestreview-1224412474

File: gateway/ln-gateway/src/rpc/mod.rs [hunk 2]
```diff
@@ -34,11 +35,20 @@ impl GatewayRpcSender {
     {
         let (sender, receiver) = oneshot::channel::<Result<R::Response>>();
         let msg = message.to_enum(sender);
-        self.sender
-            .send(msg)
+
+        if let Err(e) = self.sender.send(msg).await {
+            error!("Failed to send message over channel: {}", e);
+            return Err(e.into());
```
Comment (dpc) [inline gateway/ln-gateway/src/rpc/mod.rs:41]: `Err(e)?` `?` does a free `Into::into`.

File: gateway/ln-gateway/src/rpc/mod.rs (hunk 1 above)
Comment (dpc) [reply gateway/ln-gateway/src/rpc/mod.rs:51]: Superficially `.map_err(Into::into)` might be slightly better.

But maybe:

```rust
Ok(receiver.await.unwrap_or_else(...)?)
```

instead?

**Pull request merged**